src/devices/src/legacy/i8042.rs

```diff
@@ -0,0 +1,38 @@
+use std::convert::TryInto;
+use vm_device::bus::{PioAddress, PioAddressOffset};
+use vm_device::MutDevicePio;
+use vm_superio::I8042Device;
+
+use utils::debug;
+
+use super::EventFdTrigger;
+
+//Offset of the command register, for write accesses (port 0x64).
+const COMMAND_OFFSET: u8 = 4;
+
+// Reset CPU command.
+const CMD_RESET_CPU: u8 = 0xFE;
+
+pub struct I8042Wrapper(pub I8042Device<EventFdTrigger>);
+
+impl MutDevicePio for I8042Wrapper {
+    fn pio_read(&mut self, _base: PioAddress, offset: PioAddressOffset, data: &mut [u8]) {
+        match offset.try_into() {
+            Ok(offset) => {
+                let read_data = self.0.read(offset);
+                data[0] = read_data;
```

Comment:
you can assign here `self.0.read(offset)`, and not use the auxiliary variable `read_data`.

Reply:
We should also check the size of `data`, similarly to what we do for `write`.